src/components/list/SearchableListItem.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC, PropsWithChildren } from 'react';
+
+import { RangeTuple } from 'fuse.js';
+
+import Checkbox from '../checkbox';
+import Box from '../box/index';
+import { SearchableListItemLabel } from './SearchableListItemLabel';
+import ListItem from './ListItem';
+
+export type ListItemType = {
+  label: string;
+  value?: string;
+  facet?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onClick: (event: any) => void;
+  isSelected: boolean;
+  showDivider?: boolean;
+  showChevron?: boolean;
+  isCheckbox?: boolean;
+  isIndeterminate?: boolean;
+  highlightIndices?: readonly RangeTuple[];
+  paddingLeft?: string;
+};
+
+export const SearchableListItem: FC<PropsWithChildren<ListItemType>> = ({
+  label,
+  value,
+  facet,
+  onClick,
+  isSelected,
+  showChevron = true,
+  highlightIndices = [],
+  isCheckbox = false,
+  isIndeterminate = false,
+  children,
+}) => {
+  const labelProps = {
+    label,
+    isSelected,
+    showChevron,
+    highlightIndices,
+    isCheckbox,
+    facet,
+  };
+
+  return (
+    <ListItem css={{ breakInside: 'avoid' }}>
+      {isCheckbox ? (
```

Comment:
This makes me a bit uneasy. Should we leverage `as` and restrict it to `checkbox` and `button`?

Reply:
I'm not sure I can follow, what makes you uneasy? The button and checkbox do not have the same interface and we would end up defining the props specific and shared

Reply:
Regarding leveraging as and casting the same element I did not see a way to make that work with how the chakra implements the checkbox component. Let me know if I missed something (preferably with a tested suggestion as it took a fair bit to get the interface to this state) I did refactor the interface to take either or props how the makes it more clear/ less error-prone